### PR TITLE
feat: qwik-query package

### PR DESCRIPTION
**Pull request by gioboa**
This pull request adds Qwik adapter for TanStack Query.

I covered:

useQuery
useInfiniteQuery
userQueries
useIsFetching
useMutation
useIsMutating

It's draft because I need to cover these things

- TypeScript types
- Few tests
- Basic examples

To avoid a huge PR with too many lines of code these tasks will be in future PRs
- Docs and readme
- More tests
- More Examples

### CODE REVIEW

File: examples/qwik/basic/package.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    "eslint-plugin-qwik": "^1.2.19",
+    "prettier": "^3.1.0",
+    "typescript": "5.1.6",
+    "undici": "^5.28.0",
+    "vite": "^4.5.0",
+    "vite-tsconfig-paths": "^4.2.1"
+  },
+  "dependencies": {
+    "@tanstack/query-core": "latest",
+    "@tanstack/qwik-query": "file:../../../packages/qwik-query"
```
Comment (TkDodo) [inline examples/qwik/basic/package.json:43]: note: I think what we need to do is change this to a real dependency immediately after we've published one.

File: packages/qwik-query/package.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  "version": "0.0.1",
+  "description": "Primitives for managing, caching and syncing asynchronous and remote data in Qwik",
+  "main": "./lib/index.qwik.mjs",
+  "qwik": "./lib/index.qwik.mjs",
+  "types": "./lib-types/index.d.ts",
+  "exports": {
+    ".": {
+      "import": "./lib/index.qwik.mjs",
+      "require": "./lib/index.qwik.cjs",
+      "types": "./lib-types/index.d.ts"
```
Comment (TkDodo) [inline packages/qwik-query/package.json:12]: as far as I know, `types` must come first in `exports`

File: packages/qwik-query/package.json [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    "cypress": "~13.6.0",
+    "eslint-plugin-qwik": "1.2.12",
+    "eslint": "8.51.0",
+    "typescript": "5.1.6",
+    "undici": "5.28.0",
+    "vite-tsconfig-paths": "4.2.1",
+    "vite": "^4.5.0"
+  },
+  "dependencies": {
+    "@tanstack/query-core": "latest"
```
Comment (TkDodo) [inline packages/qwik-query/package.json:49]: please change this to:

```suggestion
    "@tanstack/query-core": "workspace:*"
```

`pnpm` will automatically replace this with the correct version at build time

File: packages/qwik-query/src/__test__/useInfiniteQuery.cy.tsx
Comment (TkDodo) [inline packages/qwik-query/src/__test__/useInfiniteQuery.cy.tsx:1]: we use `vitest` for all our other packages as test manager. Would it be possible to stick with that for qwik, too ?

File: packages/qwik-query/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+export * as QueryClientProvider from './QueryClientProvider'
+
+export { useQuery } from './useQuery'
+export { useQueries } from './useQueries'
+
+export { useInfiniteQuery } from './useInfiniteQuery'
+export { useIsFetching } from './useIsFetching'
+
+export { useMutation } from './useMutation'
+export { useIsMutating } from './useIsMutating'
```
Comment (TkDodo) [inline packages/qwik-query/src/index.ts:13]: v5 has a new `useMutationState` hook that you might want to implement in qwik as well

File: packages/qwik-query/src/queryOptions.ts
Comment (TkDodo) [inline packages/qwik-query/src/queryOptions.ts:1]: I think the `queryOptions` aren't exported from the `src/index.ts` file ?

File: packages/qwik-query/src/root.tsx
Comment (TkDodo) [inline packages/qwik-query/src/root.tsx:1]: why is this file necessary? It doesn't look like it's doing much

File: packages/qwik-query/src/utils.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+import { type QueryClient, dehydrate } from "@tanstack/query-core"
+
+export const queryClientState = (queryClient: QueryClient) => dehydrate(queryClient);
```
Comment (TkDodo) [inline packages/qwik-query/src/utils.ts:3]: there's a missing newline here. Please format with prettier.

**Review by TkDodo — CHANGES_REQUESTED** [review]
great work 👏 

some options typings are set to `any`, so I guess we need to figure this out before publishing it

File: scripts/config.js [hunk 6]
```diff
@@ -43,6 +43,11 @@ export const packages = [
     packageDir: 'packages/query-sync-storage-persister',
     entries: ['main', 'module', 'types'],
   },
+  {
+    name: '@tanstack/qwik-query',
+    packageDir: 'packages/qwik-query',
+    entries: ['main', 'module', 'types'],
+  },
```
Comment (TkDodo) [inline scripts/config.js:50]: @gioboa I added the necessary wiring in our config to release the package. Note that your `package.json` has no `module` entry. If that's intentional, please remove it from here. If it's not, please add it.

File: packages/qwik-query/src/root.tsx
Comment (gioboa) [reply packages/qwik-query/src/root.tsx:1]: Unfortunately this one is necessary otherwise the build fails. I will add a comment for that.

File: packages/qwik-query/src/utils.ts (hunk 5 above)
Comment (gioboa) [reply packages/qwik-query/src/utils.ts]: Done

File: packages/qwik-query/package.json (hunk 2 above)
Comment (gioboa) [reply packages/qwik-query/package.json]: done

File: scripts/config.js (hunk 6 above)
Comment (gioboa) [reply scripts/config.js:50]: I added it. thanks 👍

File: packages/qwik-query/src/__test__/useInfiniteQuery.cy.tsx
Comment (gioboa) [reply packages/qwik-query/src/__test__/useInfiniteQuery.cy.tsx:1]: unfortunately with Vitest I have some problem with Qwik, Cypress works like a charm, but I understand your point of view.

**Comment by TkDodo** [comment]
can you please update with main and fix the conflicts - we've release the angular adapter today 🎉 

let me see if I can help with the types

**Comment by gradddev** [comment]
> let me see if I can help with the types

I'm trying to help with that, but there's still a lot of work to be done: https://github.com/gioboa/query/pull/1

**Comment by n8sabes** [comment]
@gioboa, @TkDodo, et al.

This is **FANTASTIC**, great work, and very needed! Any idea when it might be ready to merge?

..sebastian

**Comment by gioboa** [comment]
@gradddev is doing a great job on the right direction. I'm looking forward to seeing the integration finalised.

**Comment by oqx** [comment]
Dude, I can't wait for this to release! I've been doing a weekly checkin since the last (now closed) PR. 🤣

File: pnpm-workspace.yaml [hunk 7]
```diff
@@ -7,4 +7,4 @@ packages:
   - 'examples/svelte/*'
   - 'examples/vue/*'
   - '!examples/vue/2*'
-  - '!examples/vue/nuxt*'
+  - '!examples/vue/nuxt*'
```
Comment (manudeli) [inline pnpm-workspace.yaml:10]:
```suggestion
  - '!examples/vue/nuxt*'
  
```

**Comment by gioboa** [comment]
I don't have the energy to finalise this PR now. I'm thinking at this integration twice a week but it's not correct to keep a stale PR in the repository. So I close this PR for now, I'm more than happy to explain the current situation of the PR to new contributions. I'm so sorry. Cheers 😔
